### ISSUE TITLE
External logger interferes the live event performance

### ISSUE THREAD

**Issue by sugitk** [ansible/awx#3474]
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME

 - API

##### SUMMARY

When external logger (like Splunk) is configured, it makes the live event performance be very slow.

##### ENVIRONMENT
* AWX version: Tower 3.4.2
* Operating System:  RHEL 7.6
* Web Browser: Chrome 73

##### ADDITIONAL INFORMATION

A Tower customer reported this issue.

When external logger (Splunk) is enabled, job output not displayed or very slow to be displayed in Web UI.  And sometimes external logging was failed, which may occur #2008 .
And this external logger is disabled, the issue was resolved.

**Comment by sugitk** [comment]
@ryanpetrello Should I merge this issue to #2008 ? 
The customer needs to raise priority to solve.

**Comment by ryanpetrello** [comment]
@sugitk this seems like a distinct issue worth tracking to me.

**Comment by mkim33** [comment]
@ryanpetrello Is there a way to raise priority of this issue please? we have strategic customer in Australia wanting to find progress and ETA on this, thank you!

**Comment by nfj25** [comment]
Hello,
I had a issue with an AWX 6.1.0 sending logs to a logstash. I started having jobs without any output whatsoever, even after the jobs finished.
After some investigation, I found some exceptions in a part of the AWX that looked like to be related with log shipping:

```
task_1       | 2019-10-31 20:04:01,297 ERROR    awx.main.utils.handlers Error sending message from TCPHandler: 110 Connection timed out
task_1       | Traceback (most recent call last):
task_1       |   File "/var/lib/awx/venv/awx/lib64/python3.6/site-packages/awx/main/utils/handlers.py", line 228, in _send
task_1       |     sok.connect((self._get_host(hostname_only=True), self.port or 0))
task_1       | TimeoutError: [Errno 110] Connection timed out
```

After some straces to the run_callback_receiver processes I notice that the processes were stalling for a few secconds in network operations with the logstash IP, confirming my suspicions. I disabled external log shipping and everything started to work again.

Probably the logstash service was having performance issues, but I was surprised that the AWX could be affected by external log performance problems. I assumed that the log shipping was asynchronous and would never impact AWX.

For now I disabled shipping job events to the external log, but that is not ideal.
